### ISSUE TITLE
False negative with protocols containins classvars

### ISSUE THREAD

**Issue by Tinche** [python/mypy#13537]
**Bug Report**

Some uses of protocols with ClassVars pass typechecking when they shouldn't. I've included a snippet with comments below.

**To Reproduce**

```python
from typing import Any, Protocol, ClassVar


class TestProtocol(Protocol):
    a: ClassVar[Any]


class TestClass:
    """Satisfies TestProtocol."""
    a: ClassVar = 1


def takes_instance(inst: TestProtocol) -> None:
    inst.a
    type(inst).a

takes_instance(TestClass())  # Typechecks, as it should
takes_instance(TestClass)  # Typechecks, but it blows up at runtime
```

**Expected Behavior**

The last line should not type-check.

**Actual Behavior**

The last line type-checks.

**Your Environment**

- Mypy version used: `master` branch, commit `09b8b550abe3982726047cc0ed68a9b6fa91745c`
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.10.1
- Operating system and version: macOs

**Comment by ilevkivskyi** [comment]
Hm, yeah, I think I missed this case. Should be a trivial fix.

**Comment by Tinche** [comment]
Thanks!
